src/python/pants/testutil/rule_runner.py

```diff
@@ -142,9 +143,17 @@ def target_types(self) -> FrozenOrderedSet[Type[Target]]:
         return self.build_config.target_types
 
     def request(self, output_type: Type[_O], inputs: Iterable[Any]) -> _O:
-        result = assert_single_element(
-            self.scheduler.product_request(output_type, [Params(*inputs)])
-        )
+        # TODO: Update all callsites to pass this explicitly via session values.
+        session = self.scheduler
+        for value in inputs:
+            if type(value) == OptionsBootstrapper:
+                session = self.scheduler.scheduler.new_session(
+                    build_id="buildid_for_test",
+                    should_report_workunits=True,
+                    session_values=SessionValues({OptionsBootstrapper: value}),
+                )
```

Comment:
Note: because this hack is quite tiny, I punted on actually changing the `TestBase` and `RuleRunner` APIs yet to allow for passing/setting `SessionValues`. If folks are ok with doing this as a followup, that would be my preference.

Reply:
What's the plan for when to followup? My concern is our documentation; we have users who are starting to write tests for their plugins. We say it's Alpha and we'll make changes, but still helpful to fix before too much code is written.

Fwit, I'm happy to help with this fix.